tsinfer/formats.py

```diff
@@ -2480,6 +2480,127 @@ def insert_proxy_samples(
+            ``length_multiplier''. Please ensure that the time value
+            provided matches the units used in the :class:`.AncestorData' file, i.e.
+            if your ancestors do not have site times specified, ``upper_time_bound''
+            should be between 0 and 1.
+        :param float length_multiplier: A multiplier for the length of the longest
+            ancestor between ``lower_time_bound'' and ``uppper_time_bound'', i.e.
+            if the longest ancestor in the specified range is 1 megabase, a
+            ``length_multiplier'' of 2 creates a maximum length of 2
+            megabases.
+        :param \\**kwargs: Further arguments passed to the :func:`AncestorData.copy`
+            when creating the new :class:`AncestorData` instance which will be returned.
+
+        :return: A new :class:`.AncestorData` object.
+        :rtype: AncestorData
+        """
+        self._check_finalised()
+        if upper_time_bound <= 0 or lower_time_bound <= 0:
+            raise ValueError("Time bounds cannot be negative")
+        if length_multiplier <= 0:
+            raise ValueError("Length multiplier cannot be negative")
+        if upper_time_bound < lower_time_bound:
+            raise ValueError("Upper bound must be >= lower bound")
+        sites = self.sites_position[:]
+        start = self.ancestors_start[:]
+        end = self.ancestors_end[:]
+        time = self.ancestors_time[:]
+        focal_sites = self.ancestors_focal_sites[:]
+        haplotypes = self.ancestors_haplotype[:]
+        if upper_time_bound > np.max(time) or lower_time_bound > np.max(time):
+            raise ValueError("Time bounds cannot be greater than older ancestor")
+
+        truncated = self.copy(**kwargs)
+        anc_in_bound = np.logical_and(
+            self.ancestors_time[:] >= lower_time_bound,
+            self.ancestors_time[:] <= upper_time_bound,
+        )
+        if np.sum(anc_in_bound) == 0:
+            raise ValueError("No ancestors in time bound")
+        max_length = length_multiplier * np.max(self.ancestors_length[:][anc_in_bound])
+        if max_length > self.sequence_length:
+            raise ValueError(
+                r"Maximum length of haplotype multiplied by length_multiplier is {},"
+                "which is greater than sequence length ({}). Try a smaller"
+                "length_multplier".format(max_length, self.sequence_length)
+            )
+
+        for anc in self.ancestors():
+            if anc.time > upper_time_bound and len(anc.focal_sites) > 0:
+                if sites[anc.end - 1] - sites[anc.start] > max_length:
+                    left_focal_pos = sites[np.min(anc.focal_sites)]
+                    right_focal_pos = sites[np.max(anc.focal_sites)]
+                    if right_focal_pos - left_focal_pos > max_length:
+                        logger.debug(
+                            "Ancestor contains focal sites > max_length "
```

Comment:
Looks like you wanted this to be an f-string? Not sure how helpful this debug message is going to be TBH, may just end up being confusing.